src/civic_canister_backend/src/consent_message.rs

```diff
@@ -79,38 +70,27 @@ async fn vc_consent_message(
     req: Icrc21VcConsentMessageRequest,
 ) -> Result<Icrc21ConsentInfo, Icrc21Error> {
     get_vc_consent_message(
-        &req.credential_spec,
         &SupportedLanguage::from(req.preferences),
     )
 }
 
 /// Retrieve the consent message for the given credential type and language.
 fn get_vc_consent_message(
-    credential_spec: &CredentialSpec,
     language: &SupportedLanguage,
 ) -> Result<Icrc21ConsentInfo, Icrc21Error> {
-    render_consent_message(credential_spec, language).map(|message| Icrc21ConsentInfo {
+    render_consent_message(language).map(|message| Icrc21ConsentInfo {
         consent_message: message,
         language: format!("{}", language),
     })
 }
 
 /// Show the consent message with any arguments 
 fn render_consent_message(
-    credential_spec: &CredentialSpec,
     language: &SupportedLanguage,
 ) -> Result<String, Icrc21Error> {
-    let credential_type = match verify_credential_spec(credential_spec) {
-        Ok(credential_type) => credential_type,
-        Err(err) => {
-            return Err(Icrc21Error::UnsupportedCanisterCall(Icrc21ErrorInfo {
-                description: err,
-            }));
-        }
-    };
     let template = CONSENT_MESSAGE_TEMPLATES
         .get(&(
-            CredentialTemplateType::from(&credential_type),
+            CredentialTemplateType::Credential,
```

Comment:
I dont love the idea of having user-readable consent messages hard-coded in the canister. Especially if i18n has to be handled there, as it means each new supported language is a canister change. The canister should be a data store and permissions layer only. Is that something we can delegate to the UI?

Reply:
As discussed in our meet.